lib/Server.js

```diff
@@ -40,6 +41,16 @@ const createCertificate = require('./utils/createCertificate');
 const validateOptions = require('schema-utils');
 const schema = require('./options.json');
 
+// Workaround for node ^8.6.0, ^9.0.0
+// DEFAULT_ECDH_CURVE is default to prime256v1 in these version
+// breaking connection when certificate is not signed with prime256v1
+// change it to auto allows OpenSSL to select the curve automatically
+// See https://github.com/nodejs/node/issues/16196 for more infomation
+const verMatch = process.version.match(/^v(\d+).(\d+)/);
```

Comment:
Is the version check required here or could this be defaulted to `'auto'` in general? E.g was this being 'reverted' for `node >= 10.0.0` for some reason? If so, why?

```suggestion
const version = process.version.match(/^v(\d+).(\d+)/);
```

Reply:
`node` v9.0.0 is EOL

Reply:
Just saw https://github.com/webpack/webpack-dev-server/pull/1531#issuecomment-430943980 Hmm....

Reply:
@michael-ciniawsky `node@8` also affected

Reply:
```js
const version = process.version

if (parseFloat(version.slice(1)) >= 8.6) {
  ...
}
```

Reply:
@michael-ciniawsky `node@10` is not affected

Reply:
Thanks for guiding, I've update the code and please take a look again.

There is another place using regex to match the version, I've changed it too